group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroupState.java

```diff
@@ -114,15 +115,20 @@ public enum ClassicGroupState {
         DEAD.addValidPreviousStates(STABLE, PREPARING_REBALANCE, COMPLETING_REBALANCE, EMPTY, DEAD);
     }
 
-    ClassicGroupState(String name) {
+    ClassicGroupState(String name, String lowerCaseName) {
```

Comment:
nit: I wonder if we could remove the `lowerCaseName` parameter and lower case the name in the constructor. Would it work?

Reply:
Done.